src/amoc_throttle/amoc_throttle_process.erl

```diff
@@ -205,21 +206,23 @@ inc_n(#state{n = N, max_n = MaxN} = State) ->
     NewN = N + 1,
     case MaxN < NewN of
         true ->
-            PrintableState = printable_state(State),
-            ?LOG_ERROR("~nthrottle process ~p: invalid N (~p)~n", [self(), PrintableState]),
+            internal_event(<<"throttle proccess has invalid N">>, State),
             State#state{n = MaxN};
         false ->
             State#state{n = NewN}
     end.
 
-log_state(Msg, State) ->
+internal_event(Msg, #state{name = Name} = State) ->
     PrintableState = printable_state(State),
-    ?LOG_DEBUG("~nthrottle process ~p: ~s (~p)~n", [self(), Msg, PrintableState]).
+    amoc_telemetry:execute_msg_to_self(
+      [throttle, process], #{printable_state => PrintableState, name => Name}, Msg);
+internal_event(Msg, Name) when is_atom(Name) ->
```

Comment:
you can rename `log_state` into `internal_event` but it doesn't change semantics, these function should be used for debugging purpuses only. and that's the reason why `log_state` is using `?LOG_DEBUG` macro. all other (?LOG_ERROR) places are real errors, it means that amoc_throttle configuration is done improperly.